docs/solver-pipeline.md

```diff
@@ -0,0 +1,111 @@
+---
+layout: default
+title: Solver Pipeline
+subtitle: Overview of solver pipeline and related command-line options
+slug: solver-pipeline
+---
+
+{:.toc}
+Contents
+{:.toc__title .no_toc}
+* Table of contents placeholder
+{:.toc__list .list-anchor}
+{:toc}
+
+# The solver pipeline
+
+KLEE has a number of command line options that control the solver pipeline that
+is shared both by the `klee` and `kleaver` tools.
+
+The solver pipeline is implemented internally using the decorator design pattern.
+Thus when a solver query is issued it travels through a nested stack of solvers
+and may eventually call the underlying SMT solver (core solver).
+
+The reasons for this pipe line architecture are:
+
+1. Allows transforming the solver query into another form that is hopefully
+   faster for the core solver to solve (e.g. the Independence solver).
+2. Allows querying the core solver to be avoided in some cases (e.g. the
+   Counter example caching solver).
```

Comment:
I prefer counter example (although personally I strongly dislike the use of "counter example" in KLEE's code and design. It confuses a lot of things).